code/modules/mob/living/simple_animal/abnormality/zayin/pile_of_mail.dm

```diff
@@ -0,0 +1,303 @@
+	if(cooldown > world.time)
+		to_chat(user,"<span class='warning'>You realized you have made a grave mistake as envelopes start flying out of the mailbox towards you.</span>")
+		user.Stun(10 SECONDS)
+		var/letterssave = list()
+		for(var/i = 1 to 4)
+			var/obj/item/mailpaper/hatred/MAIL = new(get_turf(src))
+			MAIL.throw_at(user, 4, 1, src, spin = FALSE, gentle = TRUE, quickstart = FALSE)
+			letterssave += MAIL
+			sleep(1 SECONDS)
+		user.gib()
+		sleep(5 SECONDS)
+		for(var/obj/item/mailpaper/MAIL in letterssave)
+			qdel(MAIL)
+		return
+
+	cooldown = world.time + cooldown_time
+
+	playsound(get_turf(src), 'sound/abnormalities/mailpile/gotmail.ogg', 50, 1)
+
+	if(prob(25))
+		var/obj/item/parcelself/PARCEL = new(get_turf(src),user)
+		PARCEL.throw_at(user, 4, 1, src, spin = FALSE, gentle = TRUE, quickstart = FALSE)
+	else
+		var/obj/item/mailpaper/junk/MAIL = new(get_turf(src))
+		MAIL.throw_at(user, 4, 1, src, spin = FALSE, gentle = TRUE, quickstart = FALSE)
+	return
+
+/mob/living/simple_animal/hostile/abnormality/mailpile/proc/Delivery(mob/living/carbon/human/user, work_type, pe, work_time)
+	if(prob(60))
+		to_chat(user,"<span class='notice'>It seems the pile did not find a message for you.</span>")
+		return
+	playsound(get_turf(src), 'sound/abnormalities/mailpile/gotmail.ogg', 50, 1)
+
+	to_chat(user,"<span class='notice'>One of the letters from the pile slowly waves in the air.</span>")
+	var/obj/item/mailpaper/MAIL
+	if(prob(20))
+		if(prob(10))
+			MAIL = new /obj/item/mailpaper/pipebomb(get_turf(src))
+		else
+			MAIL = new /obj/item/mailpaper/junk(get_turf(src))
+	else
+		switch(work_type)
+			if(ABNORMALITY_WORK_INSTINCT)
+				MAIL = new /obj/item/mailpaper/instinct(get_turf(src),user)
+			if(ABNORMALITY_WORK_INSIGHT)
+				MAIL = new /obj/item/mailpaper/insight(get_turf(src),user)
+			if(ABNORMALITY_WORK_ATTACHMENT)
+				MAIL = new /obj/item/mailpaper/attachment(get_turf(src),user)
+	MAIL.throw_at(user, 4, 1, src, spin = FALSE, gentle = TRUE, quickstart = FALSE)
+	return
+//players get a parcel / letters for completing works
+/mob/living/simple_animal/hostile/abnormality/mailpile/PostWorkEffect(mob/living/carbon/human/user, work_type, pe, work_time, canceled)
+	. = ..()
+	if(work_type == ABNORMALITY_WORK_REPRESSION)
+		DeliveryRepress(user,work_type,pe)
+	return
+
+/mob/living/simple_animal/hostile/abnormality/mailpile/NeutralEffect(mob/living/carbon/human/user, work_type, pe)
+	. = ..()
+	Delivery(user,work_type,pe)
+	return
+
+/mob/living/simple_animal/hostile/abnormality/mailpile/SuccessEffect(mob/living/carbon/human/user, work_type, pe)
+	. = ..()
+	Delivery(user,work_type,pe)
+	return
+
+//what the fuck is this
+/obj/effect/letters_flow
+	name = "Storm of letters"
+	desc = "The storm rages on."
+	icon = 'ModularTegustation/Teguicons/32x32.dmi'
+	icon_state = "letters"
+	layer = ABOVE_MOB_LAYER
+
+/obj/effect/temp_visual/maildecal
+	name = "A letter"
```

Comment:
```suggestion
	name = "letter"
```
`a`/`the` are added automatically by BYOND. Otherwise, you'd get either `That's A letter.` or `That's an a letter.` depending on if you fix the capitalization error.